src/composeWithMongooseDiscriminators.js

```diff
@@ -0,0 +1,379 @@
+    // I guess under the hud, graphql-compose shakes it off.
+    this.GQC.Query.addFields({
+      [`${this.getTypeName()[0].toLowerCase() +
+        this.getTypeName().substr(1)}One`]: this.getResolver('findOne')
+        .clone({ name: 'GenericOne' })
+        .setType(this.getType()),
+    });
+
+    // recompose Base Resolvers
+    recomposeBaseResolvers(this);
+  }
+
+  getOpts(): Options {
+    return this.opts;
+  }
+
+  getGQC(): SchemaComposer<any> {
+    return this.GQC;
+  }
+
+  getDKey(): string {
+    return this.discriminatorKey;
+  }
+
+  getDKeyETC(): EnumTypeComposer {
+    return this.DKeyETC;
+  }
+
+  getDBaseName(): string {
+    return this.modelName;
+  }
+
+  getDInterface(): GraphQLInterfaceType {
+    return this.DInterface;
+  }
+
+  hasChildDTC(DName: string): boolean {
+    return !!this.CDTCs.find(ch => ch.getTypeName() === DName);
+  }
+
+  // add fields only to DInterface, baseTC, childTC
+  addDFields(newDFields: ComposeFieldConfigMap<any, any>): this {
+    super.addFields(newDFields);
+
+    for (const CDTC of this.CDTCs) {
+      CDTC.addFields(newDFields);
+    }
+
+    return this;
+  }
+
+  extendDField(
+    fieldName: string,
+    partialFieldConfig: ComposePartialFieldConfigAsObject<any, any>
+  ): this {
+    super.extendField(fieldName, partialFieldConfig);
+
+    for (const CDTC of this.CDTCs) {
+      CDTC.extendField(fieldName, partialFieldConfig);
+    }
+
+    return this;
+  }
+
+  // relations with args are a bit hard to manage as interfaces i believe as of now do not
+  // support field args. Well if one wants to have use args, you setType for resolver as this
+  // this = this DiscriminantTypeComposer
+  // NOTE, those relations will be propagated to the childTypeComposers and you can use normally.
+  // FixMe: Note, You must use this function after creating all discriminators
+  addDRelation(fieldName: string, relationOpts: RelationOpts<any, any>): this {
+    this.addRelation(fieldName, relationOpts);
+
+    for (const CDTC of this.CDTCs) {
+      CDTC.addRelation(fieldName, relationOpts);
+    }
+
+    return this;
+  }
+
+  /* eslint no-use-before-define: 0 */
+  discriminator(childModel: Model, opts?: TypeConverterOpts): ChildDiscriminatorTypeComposer {
+    const childDTC = new ChildDiscriminatorTypeComposer(this, childModel, {
+      customizationOptions: opts || this.opts.customizationOptions,
+      reorderFields: this.opts.reorderFields,
+    });
+
+    this.CDTCs.push(childDTC);
+
+    return childDTC;
+  }
+}
+
+export class ChildDiscriminatorTypeComposer extends TypeComposer {
+  dName: string;
+
+  dTC: DiscriminatorTypeComposer;
+
+  constructor(dTC: DiscriminatorTypeComposer, childModel: Model, opts: Options) {
+    if (!childModel) {
+      throw Error('Please specify a childModel');
+    }
+    const childTC = composeWithMongoose(childModel, opts.customizationOptions);
+
+    super(childImplements(dTC, childTC).gqType);
+
+    // !ORDER MATTERS
```

Comment:
Why order matters?

Reply:
Actually, the recomposeChildResolvers references the DiscriminatorTC, so I felt putting recomposeChildResolvers before setting the child DiscriminatorTC reference might yield errors.
But I have found a way to avoid all those issues.